server/data/data-file/src/main/java/org/apache/james/mailrepository/file/MBoxMailRepository.java

```diff
@@ -250,15 +250,15 @@ private MimeMessage convertTextToMimeMessage(String emailBody) {
     }
 
     /**
-     * Generate a hex representation of an MD5 checksum on the emailbody
+     * Generate a hex representation of a SHA-256 checksum on the email body
      * 
      * @param emailBody
      * @return A hex representation of the text
      * @throws NoSuchAlgorithmException
      */
     private String generateKeyValue(String emailBody) throws NoSuchAlgorithmException {
-        // MD5 the email body for a reilable (ha ha) key
-        byte[] digArray = MessageDigest.getInstance("MD5").digest(emailBody.getBytes());
+        // SHA-256 the email body for a reliable key
```

Comment:
I guess we can remove that comment.

Reply:
Why not extracted the choice of the hash algorithm to a class that only do that.
So in the future dev will not have to choose with hash algorithm is the best and we will have only one place to change SHA-256 when it will be broken